cartofacade/carto_facade_test.go

```diff
@@ -129,3 +133,396 @@ func TestRequest(t *testing.T) {
 	err := cartoLib.Terminate()
 	test.That(t, err, test.ShouldBeNil)
 }
+
+func TestInitialize(t *testing.T) {
+	lib, err := NewLib(1, 1)
+	test.That(t, err, test.ShouldBeNil)
+
+	cancelCtx, cancelFunc := context.WithCancel(context.Background())
+	activeBackgroundWorkers := sync.WaitGroup{}
+
+	cfg, dir, err := GetTestConfig("mysensor")
+	algoCfg := GetTestAlgoConfig()
+	test.That(t, err, test.ShouldBeNil)
+	defer os.RemoveAll(dir)
+
+	cartoFacade := New(&lib, cfg, algoCfg)
+
+	t.Run("test Initialize - successful case", func(t *testing.T) {
```

Comment:
Don't you also need to call terminate once you call initialize, otherwise you will leak memory at the C++ level?

Reply:
Also, should this be mocking the Initialize C Function?

Reply:
Adding that in now!

Reply:
Is it possible to mock the C function, I wanted to but was not sure how to do it without the inject pattern

Reply:
not really, lets talk about this